tests/test_completer.py

```diff
@@ -0,0 +1,58 @@
+import pytest
+from kart.cli_util import OutputFormatType
+
+from kart.completer import conflict_completer, ref_completer
+
+DIFF_OUTPUT_FORMATS = ["text", "geojson", "json", "json-lines", "quiet", "html"]
+SHOW_OUTPUT_FORMATS = DIFF_OUTPUT_FORMATS
+
+
+def test_ref_completer(data_archive, cli_runner):
+    with data_archive("points") as _:
+        r = cli_runner.invoke(["checkout", "-b", "one"])
+        assert r.exit_code == 0, r.stderr
+        r = cli_runner.invoke(["checkout", "-b", "two"])
+        assert r.exit_code == 0, r.stderr
+
+        assert ref_completer() == [
+            "one",
+            "two",
+            "main",
+            "refs/heads/one",
+            "refs/heads/two",
+            "refs/heads/main",
+        ]
+        assert ref_completer(incomplete="r") == [
+            "refs/heads/one",
+            "refs/heads/two",
+            "refs/heads/main",
+        ]
+
+
+def test_conflict_completer(data_archive, cli_runner):
+    with data_archive("conflicts/points.tgz") as _:
+        r = cli_runner.invoke(["merge", "theirs_branch"])
+        assert r.exit_code == 0, r.stderr
+        assert conflict_completer() == [
+            "nz_pa_points_topo_150k:feature:3",
+            "nz_pa_points_topo_150k:feature:4",
+            "nz_pa_points_topo_150k:feature:5",
+            "nz_pa_points_topo_150k:feature:98001",
+        ]
+        assert conflict_completer(incomplete="nz_pa_points_topo_150k:feature:9") == [
```

Comment:
It's hard to try this out with conflicts (because you first have to create a conflict) but I think the same completer could be applied to `log` for the paths. (e.g. `kart log nz_primary_parcels:feature:8281752` gives you a log of a particular feature)

Unfortunately those aren't parsed easily since some knowledge of the repo is required to distinguish them from refnames. [code](https://github.com/amal-thundiyil/kart/blob/d5bdf681797e10b1b299f04e4b34c0422492d839/kart/log.py#L372-L373)